src/battle/overlay_12_0224E4FC.c

```diff
@@ -6478,3 +6488,435 @@ BOOL CheckStatusEffectsSubstitute(BATTLECONTEXT *ctx, int battlerId, int status)
+    }
+    
+    GF_ASSERT(FALSE);
+    return 0;
+}
+
+void ov12_022582B8(BattleSystem *bsys, BATTLECONTEXT *ctx, int var, int battlerId, int data) {
+    switch (var) {
+    case 0:
+        ctx->fieldSideConditionFlags[BattleSystem_GetFieldSide(bsys, battlerId)] = data;
+        break;
+    case 1:
+        ctx->fieldSideConditionData[BattleSystem_GetFieldSide(bsys, battlerId)].mistTurns = data;
+        break;
+    case 2:
+        ctx->selectedMonIndex[battlerId] = data;
+        break;
+    case 3:
+        ctx->totalTurns = data;
+        break;
+    case 9:
+        ctx->unk_334.unk9C = data;
+        break;
+    case 11:
+        ctx->unk_334.unkA4[battlerId] = data;
+        break;
+    default:
+        GF_ASSERT(FALSE);
+    }
+}
+
+extern const int ov12_0226CDCC[145];
+
+static int ov12_02258348(BATTLECONTEXT *ctx, int statChangeType, u32 flag) {
+    ctx->statChangeType = statChangeType;
+    ctx->statChangeParam = flag & (0x7FFFFF);
+    ctx->statChangeFlag = flag & (0xFF800000);
+    
+    if (flag & (1 << 30)) {
+        ctx->battlerIdStatChange = ctx->battlerIdAttacker;
+    } else if (flag & (1 << 31)) {
+        ctx->battlerIdStatChange = ctx->battlerIdTarget;
+    } else if ((flag & (1 << 29)) || (flag & (1 << 28))) {
+        ctx->battlerIdStatChange = BATTLER_PLAYER;
+    }
+    
+    GF_ASSERT(NELEMS(ov12_0226CDCC) > (flag & (0x7FFFFF)));
+    
+    return ov12_0226CDCC[flag & (0x7FFFFF)];
+}
+
+static int ov12_022583B4(BATTLECONTEXT *ctx, int battlerId, int typeEffectiveness, int damage, int moveDamage, u32 *flag) {
+    if (!(ctx->linkStatus & (1 << 11)) && !(ctx->linkStatus & (1 << 15)) && damage) {
+        damage = DamageDivide(damage * typeEffectiveness, 10);
+    }
+    
+    switch (typeEffectiveness) {
+    case TYPE_MUL_NO_EFFECT:
+        *flag |= MOVE_STATUS_NO_EFFECT;
+        *flag &= ~MOVE_STATUS_NOT_VERY_EFFECTIVE;
+        *flag &= ~MOVE_STATUS_SUPER_EFFECTIVE;
+        break;
+    case TYPE_MUL_NOT_EFFECTIVE:
+        if (moveDamage) {
+            if (*flag & MOVE_STATUS_SUPER_EFFECTIVE) {
+                *flag &= ~MOVE_STATUS_SUPER_EFFECTIVE;
+            } else {
+                *flag |= MOVE_STATUS_NOT_VERY_EFFECTIVE;
+            }
+        }
+        break;
+    case TYPE_MUL_SUPER_EFFECTIVE:
+        if (moveDamage) {
+            if (*flag & MOVE_STATUS_NOT_VERY_EFFECTIVE) {
+                *flag &= ~MOVE_STATUS_NOT_VERY_EFFECTIVE;
+            } else {
+                *flag |= MOVE_STATUS_SUPER_EFFECTIVE;
+            }
+        }
+        break;
+    }
+    
+    return damage;
+}
+
+static int ov12_02258440(BATTLECONTEXT *ctx, int moveNo) {
+    switch (ctx->unk_334.moveData[moveNo].effect) {
+    case 26:
+    case 39:
+    case 75:
+    case 145:
+    case 151:
+    case 155:
+    case 255:
+    case 256:
+    case 263:
+    case 273:
+        return (ctx->linkStatus & (1 << 9));
+    }
+    
+    return TRUE;
+}
+
+static u8 ov12_022584AC(BATTLECONTEXT *ctx, int battlerId, int var) {
```

Comment:
get type func

Reply:
oops I literally had it queued for the replace function as `GetMoveType` but forgot to actually run the script

Reply:
aye aye captin